### ISSUE TITLE
backward loss

### ISSUE THREAD

**Issue by jaekookang** [jaekookang/invertible_neural_networks#1]
The code template in `example_gaussian_mixture.ipynb` used MSE loss instead of MMD loss for the backward loss (as of 2020-11-29). The reason for using the MSE loss was just to accelerate training. The original authors (Ardizzone et al) used the MMD loss. It takes a bit more time for the loss to decrease to the same level as in the MSE loss case. 

But as Chris pointed out, the MMD loss facilitates the learning of the probability distribution instead of simply learning the data points, which is also what the original authors intended. I changed the code accordingly, but without guarantee. You may change this code for your own purpose and I'd appreciate your feedback if any. Also, I also appreciate Ardizzone et al for their original work on invertible neural networks.

Attached is just a simple comparison between the use of MSE and the MMD for the backward loss. You can see that with the same amount of training (1000 epochs; 3 min 18s on Colab), the MSE loss converges quickly than the MMD loss. However, the expressivity of the trained model might be not equivalent to the MMD. You can simply train more with the MMD loss, I guess.

### Use of MSE (backward loss)
![backward_MSE_loss](https://user-images.githubusercontent.com/18749927/114064573-c808b780-9867-11eb-876c-05c48458addb.png)
![backward_MSE_out](https://user-images.githubusercontent.com/18749927/114064584-c939e480-9867-11eb-9cba-8aa34a767d73.png)

### Use of MMD (backward loss)
![backward_MMD_loss](https://user-images.githubusercontent.com/18749927/114064585-c939e480-9867-11eb-809a-0c269ff6fc7f.png)
![backward_MMD_out](https://user-images.githubusercontent.com/18749927/114064587-ca6b1180-9867-11eb-91a8-d88a7e98f0fb.png)
